gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemIntegrationTest.java

```diff
@@ -1509,6 +1510,155 @@ public void testThreadTraceEnabledRename() throws Exception {
     assertThat(ghfs.exists(dest)).isTrue();
   }
 
+  @Test
+  public void testHnBucketNonRecursiveDeleteOperation() throws Exception {
+    String bucketName = this.gcsiHelper.getUniqueBucketName("hn");
+    GoogleHadoopFileSystem googleHadoopFileSystem = new GoogleHadoopFileSystem();
+    URI initUri = new URI("gs://" + bucketName);
+    Configuration config = loadConfig();
+    config.setBoolean("fs.gs.hierarchical.namespace.folders.enable", true);
+    googleHadoopFileSystem.initialize(initUri, config);
+
+    GoogleCloudStorage theGcs = googleHadoopFileSystem.getGcsFs().getGcs();
+    theGcs.createBucket(
+        bucketName, CreateBucketOptions.builder().setHierarchicalNamespaceEnabled(true).build());
+
+    assertThat(theGcs.isHnBucket(new Path(initUri.toString() + "/").toUri())).isTrue();
+    try {
+      googleHadoopFileSystem.mkdirs(new Path("A/"));
+      assertThrows(
+          "Cannot delete a non-empty directory",
+          java.nio.file.DirectoryNotEmptyException.class,
+          () -> googleHadoopFileSystem.delete(new Path("gs://" + bucketName + "/"), false));
+
+      // delete /A/ non recursively
+      googleHadoopFileSystem.delete(new Path("gs://" + bucketName + "/A/"), false);
```

Comment:
where are we validating that the folder is deleted?

Reply:
They way validation is done that if /A/ wasn't deleted properly, then line 1540 will fail, ie, we cannot rename some arbitrary directory to /A/

Reply:
I am not sure if this it the right way. Why cant we query and check?